.ci/CI/src/Bootloader.jl

```diff
@@ -0,0 +1,380 @@
+    dev_package_path::AbstractString,
+    git_ci_tools_url::String,
+    git_ci_tools_branch::String,
+)
+    job_yaml = get_normal_unit_test(
+        "1",
+        target_branch,
+        dev_package_name,
+        dev_package_version,
+        dev_package_path,
+        git_ci_tools_url,
+        git_ci_tools_branch,
+    )
+    job_yaml["image"] = "debian:bookworm-slim"
+
+    if !haskey(job_yaml, "variables")
+        job_yaml["variables"] = Dict()
+    end
+    job_yaml["variables"]["JULIA_DONWLOAD"] = "/julia/download"
+    job_yaml["variables"]["JULIA_EXTRACT"] = "/julia/extract"
+
+    job_yaml["before_script"] = [
+        "apt update && apt install -y wget",
+        "mkdir -p \$JULIA_DONWLOAD",
+        "mkdir -p \$JULIA_EXTRACT",
+        "if [[ \$CI_RUNNER_EXECUTABLE_ARCH == \"linux/arm64\" ]]; then
+  wget https://julialangnightlies-s3.julialang.org/bin/linux/aarch64/julia-latest-linux-aarch64.tar.gz -O \$JULIA_DONWLOAD/julia-nightly.tar.gz
+elif [[ \$CI_RUNNER_EXECUTABLE_ARCH == \"linux/amd64\" ]]; then
+  wget https://julialangnightlies-s3.julialang.org/bin/linux/x86_64/julia-latest-linux-x86_64.tar.gz -O \$JULIA_DONWLOAD/julia-nightly.tar.gz
+else
+  echo \"unknown runner architecture -> \$CI_RUNNER_EXECUTABLE_ARCH\"
+  exit 1
+fi",
+        "tar -xf \$JULIA_DONWLOAD/julia-nightly.tar.gz -C \$JULIA_EXTRACT",
+        # we need to search for the julia base folder name, because the second part of the name is the git commit hash
+        # e.g. julia-b0c6781676f
+        "JULIA_EXTRACT_FOLDER=\${JULIA_EXTRACT}/\$(ls \$JULIA_EXTRACT | grep -m1 julia)",
+        # copy everything to /usr to make julia public available
+        # mv is not possible, because it cannot merge folder
+        "cp -r \$JULIA_EXTRACT_FOLDER/* /usr",
+    ]
+
+    job_yaml["image"] = "debian:bookworm-slim"
+
+    job_yaml["allow_failure"] = true
+    return job_yaml
+end
+
+function add_integration_test_job_yaml!(
+    job_dict::Dict,
+    package_name::AbstractString,
+    package_version::AbstractString,
+    package_path::AbstractString,
+    git_ci_tools_url::String,
+    git_ci_tools_branch::String,
+)
+    if !haskey(job_dict, "stages")
+        job_dict["stages"] = []
+    end
+
+    package_infos = integTestGen.get_package_info()
+    if target_branch != "main"
+        integTestGen.extract_env_vars_from_git_message!(package_infos)
+    end
+    integTestGen.modify_package_url!(package_infos)
+
+    custom_urls = Dict{String,String}()
+    for (name, info) in package_infos
+        if info.modified_url != ""
+            custom_urls[name] = info.modified_url
+        end
+    end
+    qed_path = mktempdir(; cleanup=false)
+    compat_changes = Dict{String,String}()
+
+    pkg_tree = integTestGen.build_qed_dependency_graph!(
+        qed_path, compat_changes, custom_urls
+    )
+    depending_pkg = IntegrationTests.depending_projects(
+        package_name, collect(keys(package_infos)), pkg_tree
+    )
+
+    if !isempty(depending_pkg)
```

Comment:
Could be made more readable (reducing indentation depth) by changing this whole block to 
```Julia
if isempty(depending_pkg)
    return nothing
end
...
```